### PR TITLE
Added User Feature

### PR DESCRIPTION
**Pull request by aaronanish12**
### This pr has the following changes:
- acceptance test for various user feature 
- page object model implementation
- before/after hooks implementation

### CODE REVIEW

File: tests/acceptance/features/login.feature [hunk 1]
```diff
@@ -5,5 +5,11 @@ Feature: login
...
   Scenario: valid user login
     Given the user has navigated to the homepage
-    When the user logs in with email 'demo@example.com' and password 'demo' using the webUI
+    And user has been created using following details:
+      | username | demo |
+      | email | demo@example.com             |
+      | password | demo             |
+    When the user logs in with following details using the webUI:
+      | email | demo@example.com             |
+      | password | demo             |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/login.feature]: fix indentation here

File: package.json [hunk 2]
```diff
@@ -27,10 +27,12 @@
...
   "devDependencies": {
     "@cucumber/cucumber": "^8.0.0-rc.1",
+    "axios": "^0.24.0",
     "chromedriver": "^95.0.0",
     "geckodriver": "^2.0.4",
     "nightwatch": "^1.4.0",
     "nightwatch-api": "^3.0.1",
-    "selenium-server": "^3.141.59"
+    "selenium-server": "^3.141.59",
+    "utils": "^0.3.1"
```
Comment (kiranparajuli589) [inline package.json:35]: not required....remove this

File: package.json [hunk 3]
```diff
@@ -6,8 +6,8 @@
   "main": "server.js",
   "scripts": {
     "start": "node server.js",
-    "start:chrome-docker": "docker run --rm -d --network=host -p 4444:4444 -p 5900:5900 --shm-size='2g' --name=web-test-chrome selenium/standalone-chrome:4.1.0-prerelease-20211105",
-    "start:firefox-docker": "docker run --rm -d --network=host -p 4444:4444 -p 5900:5900 --shm-size='2g' --name=web-test-firefox selenium/standalone-firefox:4.1.0-prerelease-20211105",
+    "start:chrome-docker": " docker run -d -p 4444:4444 -p 5900:5900 -v /dev/shm:/dev/shm selenium/standalone-chrome-debug:3.141.59-oxygen",
+    "start:firefox-docker": "docker run --rm -d --network=host -p 4444:4444 -p 5900:5900 --shm-size='2g' selenium/standalone-firefox:4.1.0-prerelease-20211105",
```
Comment (kiranparajuli589) [inline package.json:10]: nice. 👍🏼

File: cucumber.conf.js [hunk 4]
```diff
@@ -7,6 +7,7 @@ setDefaultTimeout(60000)
 
 BeforeAll(async function (){
     await startWebDriver({})
+
```
Comment (kiranparajuli589) [inline cucumber.conf.js]: why extra line?

File: tests/acceptance/stepDefinitions/loginContext.js [hunk 5]
```diff
@@ -1,18 +1,45 @@
-const {Given, When, Then} = require('@cucumber/cucumber')
+const {Given, When, Then, After, Before} = require('@cucumber/cucumber')
 const {client} = require('nightwatch-api')
+const axios = require('axios')
+const utils = require("util")
+
+
+let userListUrl = "http://localhost:3001/api/users/"
```
Comment (kiranparajuli589) [inline tests/acceptance/stepDefinitions/loginContext.js]: use launch url here

File: tests/acceptance/stepDefinitions/loginContext.js [hunk 6]
```diff
@@ -1,18 +1,45 @@
...
+let userDeleteUrl = userListUrl + "%s"
+let createdUserList = []
+
+
+
+After(() => {
+    createdUserList.forEach((user) => {
+         axios.delete(utils.format(userDeleteUrl,user.id))
+            .then()
```
Comment (kiranparajuli589) [inline tests/acceptance/stepDefinitions/loginContext.js]: not required if not implementation is to be added
```suggestion
```

File: tests/acceptance/stepDefinitions/loginContext.js [hunk 7]
```diff
@@ -1,18 +1,45 @@
...
-        .setValue('input#email-login', email)
-        .setValue('input#password-login', password)
-        .click('.login-form button[type=submit]')
+Given('user has been created using following details:', async function (dataTable) {
+    const userToBeCreated = dataTable.rowsHash()
+    const response = await axios.post(userListUrl,userToBeCreated)
+    if (response){
+        createdUserList.push(response.data)
+    }
+    console.log(userToBeCreated["email"] + " is trying to login.")
```
Comment (kiranparajuli589) [inline tests/acceptance/stepDefinitions/loginContext.js]: ? 🤔

File: tests/acceptance/stepDefinitions/userContext.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
+const {Given, When, Then} = require('@cucumber/cucumber')
+const {client} = require('nightwatch-api')
+
+When('the user navigates to the homepage', function () {
+    return client.url(client.launchUrl)
+
```
Comment (kiranparajuli589) [inline tests/acceptance/stepDefinitions/userContext.js]:
```suggestion
```

**Review by kiranparajuli589 — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/user.feature [hunk 9]
```diff
@@ -0,0 +1,43 @@
+# Created by anish at 11/16/21
```
Comment (swoichha) [inline tests/acceptance/features/user.feature]: remove this commented line

File: tests/acceptance/features/user.feature [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+Feature: user
+  As a user
+  I want to create an account
+  So that i can login
+
+  Scenario: user is not logged in
+    When the user navigates to the homepage
```
Comment (SwikritiT) [inline tests/acceptance/features/user.feature]:
```suggestion
    When the user navigates to the homepage using the webUI
```

File: tests/acceptance/features/user.feature [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+      | email | demo@example.com             |
+      | password | demo             |
+    Then the user should see dashboard button
+    And the user should not see login button
+    And the user should not see signup button
+
+
+  Scenario: user tries to sign up
+    Given the user has navigated to the homepage
+    When the user signup using the following details using webUI:
```
Comment (SwikritiT) [inline tests/acceptance/features/user.feature]:
```suggestion
    When the user signs up with following details using the webUI:
```

File: tests/acceptance/features/user.feature [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+      | username | demo1             |
+      | email    | demo1@example.com |
+      | password | demo1             |
+    Then the signup modal should not be displayed
+    And the user should see dashboard button
+
+  Scenario:  user tries to sign up using invalid email
+    Given the user has navigated to the homepage
+    When the user signup using the following details using webUI:
```
Comment (SwikritiT) [inline tests/acceptance/features/user.feature]: same as above

File: tests/acceptance/stepDefinitions/loginContext.js (hunk 7 above)
Comment (aaronanish12) [reply tests/acceptance/stepDefinitions/loginContext.js]: async banakolai van va dai ? axios use garera user create garda await nagar nahune vayara Gary ho here

**Pull request merged**